flatpak-proxy.c

```diff
@@ -2452,16 +2434,13 @@ got_buffer_from_client (FlatpakProxyClient *client, ProxySide *side, Buffer *buf
       if (!update_socket_messages (side, buffer, header))
         return;
 
-      /* Make sure the client is not playing games with the serials, as that
-         could confuse us. */
-      if (header->serial <= client->last_serial)
+      if (header->serial > MAX_CLIENT_SERIAL)
         {
-          g_warning ("Invalid client serial");
+          g_warning ("Invalid client serial: Exceeds maximum value of %u", MAX_CLIENT_SERIAL);
           side_closed (side);
           buffer_unref (buffer);
           return;
         }
-      client->last_serial = header->serial;
```

Comment:
`client->last_serial` is not only used for the fake messages but also for the `GetNameOwner`/`get_owner_reply` mechanism and I think this breaks it.

Reply:
All uses of `client->last_fake_serial` happen after a `queue_fake_message` as far as I can tell. And `queue_fake_message` assigns a new `last_fake_serial` anyway. So I can't see the issue yet.

Reply:
Ah, you're absolutely right.